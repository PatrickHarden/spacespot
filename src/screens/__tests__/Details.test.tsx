import React from 'react'
import { Router, Route, Switch } from 'react-router-dom'
import { render } from '@testing-library/react'

import { createMockStore } from 'redux-test-utils'
import { Provider } from 'react-redux'
import Details from '../PropertyDetails'
import { createMemoryHistory } from 'history'

jest.mock('react-intl', () => ({
  ...jest.requireActual('react-intl'),
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

test('Shows the building details', () => {
  const testState = {}
  const store: any = createMockStore(testState)
  const container = render(
    <Provider store={store}>
      <Router history={createMemoryHistory({ initialEntries: ['/details/1'] })}>
        <Switch>
          <Route path="/details/:spaceId" component={Details} />
        </Switch>
      </Router>
      ,
    </Provider>,
  )
  expect(container.asFragment()).toMatchSnapshot()
})

test('Incorrect route', () => {
  const testState = {}
  const store: any = createMockStore(testState)
  const container = render(
    <Provider store={store}>
      <Router history={createMemoryHistory({ initialEntries: ['/details/1'] })}>
        <Details />
      </Router>
    </Provider>,
  )
  expect(container.asFragment()).toMatchSnapshot()
})
