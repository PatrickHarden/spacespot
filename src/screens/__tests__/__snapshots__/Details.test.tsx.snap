// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incorrect route 1`] = `
<DocumentFragment>
  <div
    class="sc-ipXKqB ktEwfJ"
  >
    <div
      class="sc-fONwsr fUhGlo"
    >
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Shows the building details 1`] = `
<DocumentFragment>
  <div
    class="sc-ipXKqB ktEwfJ"
  >
    <div
      class="sc-fONwsr fUhGlo"
    >
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
  ,
</DocumentFragment>
`;
